core/src/main/python/wlsdeploy/tool/deploy/applications_deployer.py

```diff
@@ -273,15 +273,15 @@ def __online_deploy_apps_and_libs(self, base_location):
 
         # shared library updated, app referenced must be stopped, redeployed, and started so stop the app first
         for app in stop_app_list:
-            self.__stop_app(app)
+            self.__stop_app(app, timeout=self.model_context.get_stop_app_timeout())
             # add the referenced app to the redeploy list
             redeploy_app_list.append(app)
             # add the referenced app to the start list
             deployed_app_list.append(app)
 
         # app is updated, it must be stopped and undeployed first
         for app in stop_and_undeploy_app_list:
-            self.__stop_app(app)
+            self.__stop_app(app, timeout=self.model_context.get_stop_app_timeout())
             self.__undeploy_app(app)
```

Comment:
timeout?

Reply:
I am unsure what your question means?

Reply:
should line 285 __undeploy_app has a timeout?

Reply:
refactored stop_app to be like other functions.